test/e2e/testing-manifests/storage-csi/external-snapshotter/csi-snapshotter/rbac-csi-snapshotter.yaml

```diff
@@ -23,9 +25,13 @@ rules:
   - apiGroups: [""]
     resources: ["events"]
     verbs: ["list", "watch", "create", "update", "patch"]
-  - apiGroups: [""]
-    resources: ["secrets"]
-    verbs: ["get", "list"]
+  # Secret permission is optional.
```

Comment:
happy to see this become optional. would be even happier if the example scoped to a particular namespace/name instead of giving global list permission

Reply:
Same here: this is just an example. I suppose we can make the example better by explaining more about the rationale behind the current content, though.

@cc @xing-yang

Reply:
I think we need the secret RBACs for the mock driver secret test: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-gce-storage-snapshot/1384443613999534080/

Reply:
can we narrowly grant to the specific secrets required by the test?

Reply:
Yes we should be able to grant it to the e2e namespace.

Reply:
Which E2E namespace? The one of the driver or where test objects are created?

Reply:
It looks like the test namespace: https://github.com/kubernetes/kubernetes/blob/6ede5ca95f78478fa627ecfea8136e0dff34436b/test/e2e/storage/csi_mock_volume.go#L1606

Reply:
A Pod running in one namespace cannot be granted access to secrets in one specific other namespace, right? If I remember correctly, the `Role` implicitly is for the namespace in which it gets created, which is the driver namespace.

Reply:
You can grant permissions to a service account in another namespace, see https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-binding-examples

The role/rolebinding have to be located in the namespace you want to grant access to, and the serviceaccount subject must be fully qualified:

```
subjects:
- kind: ServiceAccount
  name: some-service-account
  namespace: some-namespace
```

Reply:
Re-enabling secret access via ClusterRole was straight-forward (famous last words - the additional commit that I just pushed only passed local build testing). Adding a new service account and associated bindings is going to be more complicated. How important is that?

Reply:
In the interest of time, I would like to keep this PR focused on the immediate issue of updating the sidecars to use v1.CSINode (or have a new PR that only updates the sidecars). Any additional improvements to the tests, let's consider separately.

Reply:
> In the interest of time, I would like to keep this PR focused on the immediate issue of updating the sidecars to use v1.CSINode (or have a new PR that only updates the sidecars). Any additional improvements to the tests, let's consider separately.

sounds good

Reply:
https://github.com/kubernetes/kubernetes/issues/101334